python/delta/tables.py

```diff
@@ -581,6 +581,21 @@ def upgradeTableProtocol(self, readerVersion: int, writerVersion: int) -> None:
                              type(writerVersion))
         jdt.upgradeTableProtocol(readerVersion, writerVersion)
 
+    def addFeatureSupport(self, feature: str) -> None:
+        """
+        Modify the protocol to add a supported feature, and if the table does not support table
+        features, upgrade the protocol automatically. In such a case when the provided feature is
+        writer-only, the table's writer version will be upgraded to `7`, and when the provided
+        feature is reader-writer, both reader and writer versions will be upgraded, to `(3, 7)`.
+
+        See online documentation and Delta's protocol specification at PROTOCOL.md for more details.
+        """
+        jdt = self._jdt
+        if not isinstance(feature, str):
+            raise ValueError("The feature needs to be a string but got '%s'." % type(feature))
+        jdt.addFeatureSupport(feature)
```

Comment:
Just a nit, either we do self._jdt.addFeatureSupport(featureName) or move `jdt = self._jdt` as close as possible to the callsite, i.e. after `_verify_type_str`